### PR TITLE
Cancer Types Summary tab updates

### PR DESCRIPTION
**Pull request by kalletlak**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/6301

1. Show whether profiled data is present or not
2. Include gene panel data

### CODE REVIEW

**Comment by alisman** [comment]
@jjgao do you think we need to add definition for plus/minus:  - = no data available?  or is that obvious?

**Comment by jjgao** [comment]
It would be a good idea to add a tooltip, - mean "Not profiled"

**Comment by kalletlak** [comment]
@jjgao then what should be the symbol for samples profiled but not alterations?

**Comment by kalletlak** [comment]
@jjgao should we hide bars with no data. Example, hiding `Glioma` and `Gastrointestinal Stromal Tumor` here
Screen Shot 2019-07-01 at 11 24 11 AM

**Comment by jjgao** [comment]
> @jjgao then what should be the symbol for samples profiled but not alterations?

It should be +.

@jjgao should we hide bars with no data. Example, hiding Glioma and Gastrointestinal Stromal Tumor here

I think that's a good idea, but only hide bars when none of the samples are profiled in any of the profiles.

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx [hunk 1]
```diff
@@ -60,61 +76,145 @@ export function mergeAlterationDataAcrossAlterationTypes(alterationData:ICancerS
...
-        return this.bodyWidth + this.rightPadding + this.leftPadding;
+    @computed get barPlotTooltipComponent() {
+        if (!this.barPlotTooltipModel) {
+            return null;
+        } else {
+            const tooltipModel = this.barPlotTooltipModel
+            const maxWidth = 400;
+            let tooltipPlacement = (this.mousePosition.x > WindowStore.size.width - maxWidth ? "left" : "right");
+
+            return (ReactDOM as any).createPortal(
```
Comment (alisman) [inline src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: Not important but you can use <Portal> component for this

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx [hunk 2]
```diff
@@ -162,14 +257,13 @@ export class CancerSummaryChart extends React.Component<CancerSummaryChartProps,
...
-                                        x:props.x + 20 - this.scrollPane.scrollLeft,
-                                        y:props.y - 18,
+                                    self.barPlotTooltipModel = {
+                                        ...props,
                                         groupName:props.datum.x,
                                         alterationData:self.props.countsByGroup[props.datum.xKey]
                                     };
                                 } else {
-                                    self.hideTooltip();
+                                    self.barPlotTooltipModel = null;
```
Comment (alisman) [inline src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: why not just change this inside hideTooltip method?

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx [hunk 3]
```diff
@@ -245,65 +325,210 @@ export class CancerSummaryChart extends React.Component<CancerSummaryChartProps,
...
+    private barChartHeight() {
+        return 400;
+    }
 
-        // if we're not showing result broken down the alterations then we need to merge
+    @computed get scatterChartHeight() {
+        return this.scatterChartExtent;
+    }
+
+    @computed get scatterChartExtent() {
```
Comment (alisman) [inline src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: seems poorly named.  If it's only being used by the height getter above, why not just put the calculation there?

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx [hunk 4]
```diff
@@ -245,65 +325,210 @@ export class CancerSummaryChart extends React.Component<CancerSummaryChartProps,
...
-                    <div ref={(el:HTMLDivElement)=>this.scrollPane=el} style={{overflowX:'auto', overflowY:'hidden'}}>
-                    {
-                        (this.tooltipModel) && (this.buildTooltip(this.tooltipModel))
-                    }
-                    <VictoryChart width={this.width} height={this.height} theme={CBIOPORTAL_VICTORY_THEME}
-                                  padding={{bottom:this.bottomPadding, top:this.topPadding, left:this.leftPadding, right:this.rightPadding }} domainPadding={{x: 20, y: 20}}
-                                  containerComponent={<VictoryContainer containerRef={(ref: any) => this.svgContainer = ref} responsive={false}/>}
+            <div style={this.overflowStyle} className="borderedChart">
+                <div style={{ overflowX: 'auto', overflowY: 'hidden' }}>
+                    <svg
```
Comment (alisman) [inline src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: what's reason for all this extra wrapping?

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx (hunk 2 above)
Comment (kalletlak) [reply src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: it's the same statement even in that method

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx (hunk 4 above)
Comment (kalletlak) [reply src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: this is to have both bar plot and scatter plot together

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx (hunk 2 above)
Comment (alisman) [reply src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: then why not just use it?   not important, but just trying to understand

**Pull request merged**